torch/nn/functional.py

```diff
@@ -5014,6 +5014,8 @@ def multi_head_attention_forward(
     # set up shape vars
     tgt_len, bsz, embed_dim = query.shape
     src_len, _, _ = key.shape
+    assert key_padding_mask is None or key_padding_mask.dtype == torch.bool or key_padding_mask.dtype == torch.float, \
```

Comment:
Hmm, `assert` sounds like a bad mechanism for me, as they could be suppressed via a runtime settings.
It's better to raise an `AssertionError`